### PR TITLE
adds ledger utility class

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Adds Ledger Utility class. It has 4 main functions: 

1. Connect to the device 
2. Retrieve the public address
3. Retrieve the view only keys
4. Sign a message

This utility class will be used by the CLI commands which will be introduced right after this PR. 

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/utils/ledger.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    if (!response.publicAddress) {
+      this.logger.debug(`No public address returned.`)
+      this.logger.debug(response.returnCode.toString())
+      throw new Error(response.errorMessage)
+    }
+
+    return response.publicAddress.toString('hex')
+  }
+
```
Comment (jowparks) [inline ironfish-cli/src/utils/ledger.ts:68]: Why isn't this in the sdk and consumed by CLI? This could potentially be useful for other developers

Comment (patnir) [reply ironfish-cli/src/utils/ledger.ts:68]: Our philosophy with Zondax/ Chainport is that we want to keep all 3rd party stuff in the application level and not in the sdk.

**Review by patnir — COMMENTED** [review]
This is a test review

**Review by patnir — COMMENTED** [review]
This is a test review

**Review by mat-if — APPROVED** [review]

File: ironfish-cli/package.json [hunk 2]
```diff
@@ -70,6 +70,8 @@
     "axios": "1.7.2",
     "bech32": "2.0.0",
     "blessed": "0.1.81",
+    "@ledgerhq/hw-transport-node-hid": "6.28.6",
```
[the remark this answers is not in the document]
Comment (patnir) [reply ironfish-cli/package.json]: Done!

**Pull request merged**